input/pagecontent/algorithms.md

```diff
@@ -1,257 +1,189 @@
+##### De-Identifying Categorical Quasi-Identifiers
 
-Some standard substitute data have traditionally been used as missing indicators. Common examples are “John Doe” and “999999”. This approach is less desirable because of the potential for confusion and the need for special software processing. Mistaken use of such indicators has led to a variety of statistical data processing problems.
+In statistics, a categorical variable (also called qualitative variable) is a variable that can take on one of a limited, and usually fixed, number of possible values, assigning each individual or other unit of observation to a particular group or nominal category on the basis of some qualitative property.[Source: Wikipedia](https://en.wikipedia.org/wiki/Categorical_variable#cite_note-yates-1). In the context of de-identification, categorical quasi-identifiers are categorical variables. In healthcare, patient gender, diagnostic code are typical examples of categorical quasi-identifiers.
 
-Encryption and hashing have been intentionally omitted from this description. These are difficult to implement properly. They are often vulnerable to dictionary attack.
+Common techniques for de-identifying categorical variables include:
 
-#### Example
+**Generalization:** This involves reducing the granularity of data by grouping specific values into broader categories. For example, replacing the ICD-10 diagnosis code E11.2(Type 2 diabetes with kidney complications) and E11.4(Type 2 diabetes with neurological complications) with E11(Type 2 diabetes mellitus). To apply the generalization in dealing with categorical quasi-identifiers, the hierarchy definition is required. For example, in the ICD-10 Hierarchy, E11.2 and E11.4 fall under E11. Selecting the appropriate level of the generalized value needs to be carefully discussed and conformed with the intended data receivers to make sure the reduced level of the granularity of the new value is acceptable regarding the purpose of the data collection.
 
-Historically, this technique has been used for legal and governmental work when printed content is physically obscured with a black mark preventing the original content from being read.
+**Suppression:** Removing or masking highly identifiable categories (e.g., omitting rare diagnosis code).
 
-<figure>
-<img src="physically-redacted-document.png"
-style="width:3.81319in;height:2.62639in" />
-<figcaption><strong>Figure: Physically redacted USA CIA document.</strong></figcaption>
-</figure>
-<br>
+**Randomization:** Randomization techniques like permutation can be used to exchange values of categorical variables between records to obscure individual identities while maintaining overall distributions.
 
-[Source](http://upload.wikimedia.org/wikipedia/commons/0/09/Redacted_CIA_document.jpg)
+**Blanking and imputing:** Specific values that are highly identifying can be removed and replaced with imputed values.
```

Comment:
I think the previous algorithm had multiple styles of this that were more actionable and understandable than the word "imputed values". Previously we had 4 different forms of "Redaction" that applied to codes

Reply:
that was not clear from your text. So good to make more clear. Please explain the concept of imputed values, and add it to the definitions.